projects/feature-activation/0001-feature-activation.md

```diff
@@ -0,0 +1,558 @@
+
+This should not be confused with voting. When a feature is defined in the feature activation process, it is assumed that it has been agreed by the community and ideally it should be activated by the end of the process. What bit signaling does, is preventing that some feature is activated before a super majority of miners are ready to support it. It is not voting whether to accept or reject some feature. This was a problem in Bitcoin's SegWit update, where some miners used their influence (through hash rate) to delay the update, for either economical, political, or other obscure reasons. [4][5]
+
+With that being said, it is possible that a feature may or may not be activated at the end of an activation process. To preserve flexibility, that behavior will be configurable on a per-feature basis through the `activate_on_timeout` attribute, explained below. That attribute is analogous to the infamous Bitcoin LOT attribute (lock in on timeout) [6], proposed in BIP 8.
+
+#### Burying feature activations
+[Burying feature activations]: #burying-feature-activations
+
+Burying a feature activation means changing code so "history is erased", and it's not clear that a feature was activated using the activation process.
+
+It is allowed to change code that is conditional on feature activation, but **NOT** to remove a feature from the definition, as described in the [Feature and Criteria definition] section.
+
+## Feature and Criteria definition
+[Feature and Criteria definition]: #feature-and-criteria-definition
+
+The definition of new features available for feature activation should be as simple as defining the feature name and the criteria required to evaluate its activation state. Once defined, ideally, it should not be changed. The `Feature` enum will contain all features, past and future, activated or not:
+
+```python
+class Feature(Enum):
+    NANO_CONTRACTS = Criteria(name='NANO_CONTRACTS', ...)
+    SOME_NEW_OP_CODE = Criteria(name='SOME_NEW_OP_CODE', ...)
+    SOME_OTHER_FEATURE = Criteria(name='SOME_OTHER_FEATURE', ...)
+```
+
+The `Criteria` data class is used to define attributes such as the feature name and activation requirements. Each name should be unique across all features, past and future. Its definition is provided in the [Criteria configuration] section.
+
+Different features with different criteria should be configurable separately for each network, mainnet and testnet. Feature definitions should **NEVER** be removed, even after activation, as explained in the [Burying feature activations] section. This is meant to preserve history.
```

Comment:
How would we pick which features are available for each network?

Reply:
What do you mean by pick? Like, how the code picks it or how humans would select which feature to specify?

Reply:
I was asking how each network would list its available features. But, as each feature would be for only one network, it does not seem to be a problem. Make sense?

Reply:
Oh I see, makes sense. Indeed, that wasn't mentioned. As just commented [here](https://github.com/HathorNetwork/rfcs/pull/51#discussion_r1180764556), the list of features under the activation process would be defined in `HathorSettings`, for each network.

Reply:
Done in https://github.com/HathorNetwork/rfcs/pull/51/commits/7b70d95ec05dc13a16f73f4e6fef31991c960525

Reply:
Thanks!